include/pru/am572x.h

```diff
@@ -0,0 +1,769 @@
+#define PRU_INTC_HIEISR_OFFS	0x34
+#define PRU_INTC_HIDISR_OFFS	0x38
+#define PRU_INTC_GPIR_OFFS		0x80
+#define PRU_INTC_SRSR0_OFFS		0x200
+#define PRU_INTC_SRSR1_OFFS		0x204
+#define PRU_INTC_SECR0_OFFS		0x280
+#define PRU_INTC_SECR1_OFFS		0x284
+#define PRU_INTC_ESR0_OFFS		0x300
+#define PRU_INTC_ERS1_OFFS		0x304
+#define PRU_INTC_ECR0_OFFS		0x380
+#define PRU_INTC_ECR1_OFFS		0x384
+#define PRU_INTC_CMR0_OFFS		0x400
+#define PRU_INTC_CMR1_OFFS		0x404
+#define PRU_INTC_CMR2_OFFS		0x408
+#define PRU_INTC_CMR3_OFFS		0x40c
+#define PRU_INTC_CMR4_OFFS		0x410
+#define PRU_INTC_CMR5_OFFS		0x414
+#define PRU_INTC_CMR6_OFFS		0x418
+#define PRU_INTC_CMR7_OFFS		0x41c
+#define PRU_INTC_CMR8_OFFS		0x420
+#define PRU_INTC_CMR9_OFFS		0x424
+#define PRU_INTC_CMR10_OFFS		0x428
+#define PRU_INTC_CMR11_OFFS		0x42c
+#define PRU_INTC_CMR12_OFFS		0x430
+#define PRU_INTC_CMR13_OFFS		0x434
+#define PRU_INTC_CMR14_OFFS		0x438
+#define PRU_INTC_CMR15_OFFS		0x43c
+#define PRU_INTC_HMR0_OFFS		0x800
+#define PRU_INTC_HMR1_OFFS		0x804
+#define PRU_INTC_HMR2_OFFS		0x808
+#define PRU_INTC_HIPIR0_OFFS	0x900
+#define PRU_INTC_HIPIR1_OFFS	0x904
+#define PRU_INTC_HIPIR2_OFFS	0x908
+#define PRU_INTC_HIPIR3_OFFS	0x90c
+#define PRU_INTC_HIPIR4_OFFS	0x910
+#define PRU_INTC_HIPIR5_OFFS	0x914
+#define PRU_INTC_HIPIR6_OFFS	0x918
+#define PRU_INTC_HIPIR7_OFFS	0x91c
+#define PRU_INTC_HIPIR8_OFFS	0x920
+#define PRU_INTC_HIPIR9_OFFS	0x924
+#define PRU_INTC_SIPR0_OFFS		0xd00
+#define PRU_INTC_SIPR1_OFFS		0xd04
+#define PRU_INTC_SITR0_OFFS		0xd80
+#define PRU_INTC_SITR1_OFFS		0xd84
+#define PRU_INTC_HINLR0_OFFS	0x1100
+#define PRU_INTC_HINLR1_OFFS	0x1104
+#define PRU_INTC_HINLR2_OFFS	0x1108
+#define PRU_INTC_HINLR3_OFFS	0x110c
+#define PRU_INTC_HINLR4_OFFS	0x1110
+#define PRU_INTC_HINLR5_OFFS	0x1114
+#define PRU_INTC_HINLR6_OFFS	0x1118
+#define PRU_INTC_HINLR7_OFFS	0x111c
+#define PRU_INTC_HINLR8_OFFS	0x1120
+#define PRU_INTC_HINLR9_OFFS	0x1124
+#define PRU_INTC_HIER_OFFS		0x1500
+
+
+struct __PRU_INTC_REVID_S {
+	const volatile unsigned int REVISION : 32;
+};
+
+
+struct __PRU_INTC_CR_S {
+	const volatile unsigned int __reserved_1  : 1;
+	volatile unsigned int WAKEUP_MODE		  : 1;
+	volatile unsigned int NEST_MODE			  : 2;
+	volatile unsigned int PRIORITY_HOLD_MODE  : 1;
+	const volatile unsigned int __reserved_2  : 27;
+};
+
+
+struct __PRU_INTC_GER_S {
+	volatile unsigned int ENABLE_HINT_ANY		: 1;
+	const volatile unsigned int __reserved_2	: 31;
+};
+
+
+struct __PRU_INTC_GNLR_S {
+	volatile unsigned int GLB_NEST_LEVEL		: 9;
+	const volatile unsigned int __reserved_1	: 22;
+	volatile unsigned int AUTO_OVERRIDE			: 1;
+};
+
+
+struct __PRU_INTC_SISR_S {
+	volatile unsigned int STATUS_SET_INDEX		: 10;
+	const volatile unsigned int __reserved_1	: 22;
+};
+
+
+struct __PRU_INTC_SICR_S {
+	volatile unsigned int STATUS_CLR_INDEX		: 10;
+	const volatile unsigned int __reserved_1	: 22;
+};
+
+
+struct __PRU_INTC_EISR_S {
+	volatile unsigned int ENABLE_SET_INDEX		: 10;
```

Comment:
Can you please also fix this for AM3?

Reply:
Done in 9b9e13b68f93c694947aacbfe69aa683abe0f3f8